static/app/utils/profiling/profile/sampledProfile.tsx

```diff
@@ -71,17 +74,25 @@ export class SampledProfile extends Profile {
         ? sortSamples(sampledProfile)
         : stacksWithWeights(sampledProfile);
 
+    // We process each sample in the profile while maintaining a resolved stack of frames.
+    // There is a special case for GC frames where they are appended on top of the previos stack.
+    // By reusing the stack buffer, we avoid creating a new array for each sample and for GC case,
+    // also avoid re-resolving the previous stack frames as they are already in the buffer.
+
+    // If we encounter multiple consecutive GC frames, we will merge them into a single sample
+    // and sum their weights so that only one of them is processed.
+
+    // After we resolve the stack, we call appendSampleWithWeight with the stack buffer, weight
+    // and size of the stack to process. The size indicates how many items from the buffer we want
+    // to process.
+
+    const resolvedStack: Frame[] = new Array(256); // stack size limit
```

Comment:
Yeah, I wasnt sure. I'll update

Reply:
Actually @Zylphrex I'm not sure about 128, seems like most platforms have that limit, but It's not 100% (idk what is the android behavior here). Since it doesnt hurt to preallocate it a bit larger here, I'd rather err on the safer side.